diff --git a/tools/clang/tools/dxcompiler/CMakeLists.txt b/tools/clang/tools/dxcompiler/CMakeLists.txt
index 004d2e5a..7db7842d 100644
--- a/tools/clang/tools/dxcompiler/CMakeLists.txt
+++ b/tools/clang/tools/dxcompiler/CMakeLists.txt
@@ -126,7 +126,7 @@ if (MSVC)
   find_package(DiaSDK REQUIRED) # Used for constants and declarations.
 endif (MSVC)
 
-add_clang_library(dxcompiler SHARED ${SOURCES})
+add_clang_library(dxcompiler ${LSTGEXT_LIBTYPE} ${SOURCES})
 add_dependencies(dxcompiler TablegenHLSLOptions) 
 if (MSVC)
   # No DxcEtw on non-Windows platforms.
